lib/features/family/features/reflect/presentation/pages/grateful_screen.dart

```diff
@@ -42,66 +58,157 @@ class _GratefulScreenState extends State<GratefulScreen> {
+            ),
+          );
+        }
       },
+      child: BaseStateConsumer(
+        cubit: _cubit,
+        onCustom: _handleCustom,
+        onLoading: (context) => const GratefulLoading(),
+        onData: (context, uiModel) {
+          return FunScaffold(
+            withSafeArea: false,
+            appBar: FunTopAppBar(
+              title: "You're grateful for",
+              actions: [
+                LeaveGameButton(
+                  onPressed: () => const FinishReflectionDialog().show(context),
+                ),
+              ],
+            ),
+            body: Column(
+              children: [
+                GratefulAvatarBar(
+                    uiModel: uiModel.avatarBarUIModel,
+                    onAvatarTapped: _cubit.onAvatarTapped),
+                Flexible(
+                  child: RecommendationsWidget(
+                    uiModel: uiModel.recommendationsUIModel,
+                    onRecommendationChosen: (int i) {
+                      _cubit.onRecommendationChosen(i);
+                      context.pop();
+                    },
+                  ),
+                ),
+              ],
+            ),
+          );
+        },
+      ),
     );
   }
 
   void _handleCustom(BuildContext context, GratefulCustom custom) {
     switch (custom) {
       case final GratefulOpenKidDonationFlow data:
-        context.read<ProfilesCubit>().setActiveProfile(data.profile.userId);
-        Navigator.of(context).push(
-          BlocProvider(
-            create: (BuildContext context) =>
-                CreateTransactionCubit(context.read<ProfilesCubit>(), getIt()),
-            child: ChooseAmountSliderScreen(
-              onCustomSuccess: () {
-                _cubit.onDonated(data.profile);
-                context.pop();
-              },
-            ),
-          ).toRoute(context),
+        _navigateToChildGivingScreen(
+          context,
+          data.profile,
         );
       case final GratefulOpenParentDonationFlow data:
-        // TODO: Handle this case.
-        data.profile;
+        _navigateToParentGivingScreen(
+          context,
+          data.organisation,
+        );
       case GratefulGoToGameSummary():
         _navigateToSummary(context);
     }
   }
 
+  Future<void> _navigateToChildGivingScreen(
+    BuildContext context,
+    GameProfile profile,
+  ) async {
+    await context.read<ProfilesCubit>().setActiveProfile(profile.userId);
+    await Navigator.of(context).push(
+      BlocProvider(
+        create: (BuildContext context) =>
+            CreateTransactionCubit(context.read<ProfilesCubit>(), getIt()),
+        child: ChooseAmountSliderScreen(
+          onCustomSuccess: () {
+            _cubit.onDonated(profile);
+            context.pop();
+          },
+        ),
+      ).toRoute(context),
+    );
+  }
+
+  Future<void> _navigateToParentGivingScreen(
+    BuildContext context,
+    Organisation org,
+  ) async {
+    _medium.setMediumId(org.namespace);
```

Comment:
Do we need
`    await context.read<ProfilesCubit>().setActiveProfile(profile.userId);`
here?

Reply:
no, thats for kids

Reply:
the GiveBloc uses the userid from the AuthCubit, not ProfilesCubit

Reply:
Clear, ty